### PR TITLE
Add universal bound literals

### LINKED ISSUE #1158: Switch to universal bound value literals
As a prerequisite to binding values on generic type pins, we need a unified system which unambiguously maps a string to a typed data value. So instead of relying on JSON type system, let's store all bound values as strings with an enforced format.

Type | Now | Then
---- | ---- | ---
Boolean | true | "True"
Number | 42.1 | "42.1"
String | "Hello" | "\"Hello\""
Pulse | "ON_BOOT" | "On Boot"

Currently, the sole purpose of keeping bound values is passing them as is to C++ transpiler. No other logic depends on concrete values. Since it is crucial to keep backward compatibility we can adjust:

```js
Handlebars.registerHelper('cppValue', (type, value) => {...})
```

so that it can deal with the old and new form at the same time. The `type` is known, so, for example, it can interpret both `"True"` and `true` as C++ `true` if `type` is `PIN_TYPE.BOOLEAN`.

To aid a gradual switch to literals and allow binding to generics, the widgets should be changed so that they can *get* values in both the old and new form and *set* values on a commit in the new literal form only. Widgets for all four basic types are affected.

Furthermore, the string widget should be smart enough to automatically perform unquote + enquote right before commit. So a xoder who forgets to enquote manually will get the right literal automatically.

### PR DESCRIPTION
**Pull request by brusherru**
It closes #1158.
Also fixed some minor bug, that dispatches `NODE_PROPERTY_UPDATE` twice.

### CODE REVIEW

File: packages/xod-client/src/editor/components/inspectorWidgets/Widget.jsx [hunk 1]
```diff
@@ -79,15 +86,34 @@ export default function composeWidget(Component, widgetProps) {
...
-      const parsedValue = this.parseValue(this.state.value);
-      if (parsedValue !== this.parseValue(this.props.value)) {
-        this.props.onPropUpdate(
-          this.props.entityId,
-          this.props.kind,
-          this.props.keyName,
-          parsedValue
-        );
-      }
+      // Prevent of commiting widgets without changes provided by User
```
Comment (evgenykochetkov) [inline packages/xod-client/src/editor/components/inspectorWidgets/Widget.jsx]: Another thing to solve in #766 later on 😉

File: packages/xod-project/src/constants.js [hunk 2]
```diff
@@ -22,11 +22,17 @@ export const PIN_TYPE = {
   T3: 't3',
 };
 
+export const INPUT_PULSE_PIN_BINDING_OPTIONS = {
+  Never: 'Never',
```
Comment (evgenykochetkov) [inline packages/xod-project/src/constants.js]: What's wrong with all-caps keys?
IMHO, `INPUT_PULSE_PIN_BINDING_OPTIONS['On Boot']` would look a bit awkward

Comment (brusherru) [reply packages/xod-project/src/constants.js]: Yep. But in this case, we don't need to have another map that will point from 'On Boot' literal to 'ON_BOOT' key in this map. Or not? 🤔

File: packages/xod-arduino/src/templates.js [hunk 3]
```diff
@@ -114,15 +115,26 @@ Handlebars.registerHelper('global', function global(options) {
...
+        R.equals('True'),
+        R.always('true'),
+        R.always('false')
+      ),
+      [PIN_TYPE.NUMBER]: R.identity,
+      [PIN_TYPE.STRING]: R.compose(
+        cppStringLiteral,
+        R.replace(/"/g, '\\"'),
+        unquote
+      ),
```
Comment (nkrkv) [inline packages/xod-arduino/src/templates.js]: The function exploded a bit. `R.replace(/"/g, '\\"')` should semantically belong to `cppStringLiteral`.

The repetitive code for `BOOLEAN`, perhaps, better expressed with a simple ternary.

**Comment by brusherru** [comment]
@nkrkv all fixed

**Pull request merged**